File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/COMPOSICION-AGREGACION/Ejercicio1.java
import java.util.ArrayList;

class Habitacion {
    private String nombre;
    private double tamano;

    public Habitacion(String nombre, double tamano) {
        this.nombre = nombre;
        this.tamano = tamano;
    }

    public String mostrarInfo() {
        return "Habitación: " + nombre + ", Tamaño: " + tamano + " m²";
    }
}

class Casa {
    private String direccion;
    private ArrayList<Habitacion> habitaciones;

    public Casa(String direccion) {
        this.direccion = direccion;
        this.habitaciones = new ArrayList<>();
    }

    public void agregarHabitacion(Habitacion habitacion) {
        habitaciones.add(habitacion);
    }

    public void mostrarCasa() {
        System.out.println("Casa ubicada en: " + direccion);
        for (Habitacion h : habitaciones) {
            System.out.println(h.mostrarInfo());
        }
    }
}
// SIN GETTERS NI SETTERS
class TestCasa {
    public static void main(String[] args) {
        Casa casa = new Casa("Av. Siempre Viva 742");
        casa.agregarHabitacion(new Habitacion("Sala", 25));
        casa.agregarHabitacion(new Habitacion("Cocina", 12));
        casa.agregarHabitacion(new Habitacion("Dormitorio", 20));

        casa.mostrarCasa();
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/AGUILAR_MACHICADO_NILS_REINER/PRACTICA 2/COMPOSICION-AGREGACION/Ejercicio1.py
class Habitacion:
    def __init__(self, nombre, tamano):
        self.nombre = nombre
        self.tamano = tamano

    def mostrar_info(self):
        return f"Habitación: {self.nombre}, Tamaño: {self.tamano} m²"
# SIN GETTERS NI SETTERS
class Casa:
    def __init__(self, direccion):
        self.direccion = direccion
        self.habitaciones = []

    def agregar_habitacion(self, habitacion):
        self.habitaciones.append(habitacion)

    def mostrar_casa(self):
        print(f"Casa ubicada en: {self.direccion}")
        for h in self.habitaciones:
            print(h.mostrar_info())

if __name__ == "__main__":
    casa = Casa("Av. Siempre Viva 742")
    casa.agregar_habitacion(Habitacion("Sala", 25))
    casa.agregar_habitacion(Habitacion("Cocina", 12))
    casa.agregar_habitacion(Habitacion("Dormitorio", 20))
    casa.mostrar_casa()